test/Interop/Cxx/stdlib/use-std-map.swift

```diff
@@ -427,6 +427,24 @@ StdMapTestSuite.test("UnorderedMap.merge(map)") {
   expectEqual(m[3], 3)
 }
 
+StdMapTestSuite.test("Map.merge(CxxDictionary)") {
+  var m = initEmptyMap()
+  m.merge(initUnorderedMap()) { _, v1 in v1 }
```

Comment:
Could you please also add a couple of checks here?
* merging a non-empty map with another non-empty map
* merging a non-empty map with an empty map
* merging an empty map with another empty map